HW11.py

```diff
@@ -0,0 +1,129 @@
+"""
+Homework 11. Bank Card operations.
+
+# write class for bank card.
+# Class should reflect card lifecycle, card operations etc.
+# use CVV, PIN, Name, Surname , end date, card_num as initial params
+# class should have in addition to common logic some class attributes,
+# as minimum one classmethod and as minimum  one staticmethod,
+# two or more getters/setters, __str_ magic method
+# do not forget about block if __name__ == '__main__':
+# and code there to check your class written logic.
+"""
+
+import re
+from datetime import datetime as dt
+from random import SystemRandom as Sr
+
+rand = Sr()
+
+
+class CardOperations:
+    """Card operations."""
+
+    CVV_DIGITS = 3
+    card_type = 'Standard'
+
+    def __init__(self, cvv, card_num, end_date, surname, name, pin, money=0):
+        """Has params."""
+        self.__CVV = cvv if self.check_cvv(cvv) else str(rand.randint(1, 999)).zfill(3)
+        self.__PIN = pin
+        self.Name = name
+        self.Surname = surname
+        self.EndDate = end_date
+        self.__Card_num = card_num if card_num else str(rand.randint(1, 9999999999999999)).zfill(12)
+        self.Money = float(money)
+
+    def check_pin(self):
+        """Check pin code with 3 attempts."""
+        att = 2
+        while att >= 0:
+            your_pin = input('Enter PIN:')
+            if int(your_pin) == self.pin:
+                return True
+            else:
+                print(f'Error, {att} attempts left')
+                att -= 1
+        return False
+
+    def adding(self, money):
+        """Refill."""
+        self.Money += float(money)
+        return float(self.Money)
+
+    def withdraw(self, money):
+        """Withdrawal from account."""
+        if float(money) <= self.Money:
+            self.Money -= float(money)
+            return float(self.Money)
+        else:
+            raise ValueError('Not enough money')
+
+    @classmethod
+    def check_cvv(cls, cvv):
+        cvv_pat = '^/d{' + str(cls.CVV_DIGITS) + '}$'
+        return True if re.match(pattern=cvv_pat, string=str(cvv)) else False
+
+    @property
+    def cvv(self):
+        return self.__CVV
+
+    @property
+    def pin(self):
```

Comment:
please create setter for pin too. 
Setter could have such logic for example ->  when we set new value - we just ask and check previous pin and if all correct then we change existing pin